### ISSUE TITLE
Crash using JITLink on RISC-V with linker relaxation

### ISSUE THREAD

**Issue by hahnjo** [llvm/llvm-project#63671]
The following C++ snippet defines global `std::vector<int>` and pushes a single element in `main`:
```c++
#include <vector>

std::vector<int> v;

int main() {
  v.push_back(42);
  return 0;
}
```

After compiling with `clang++ -S -emit-llvm global-vector.push.cpp`, "interpretation" works fine with `lli -mattr=+d global-vector.push.ll`,  but crashes with `lli -mattr=+d,+relax global-vector.push.ll`.

I tried debugging and selectively disabling relaxation of certain instructions. It seems the problem is introduced by relaxing the call to `_ZNSt6vectorIiSaIiEE9push_backEOi` into a `jal` (because `RD = 1`), but I fail to understand why - a number of other relaxations of the same pattern work just fine.

/cc @mtvec

**Comment by mtvec** [comment]
Thanks for reporting this! Should be fixed by [D154501](https://reviews.llvm.org/D154501).

**Comment by hahnjo** [comment]
Awesome, thanks for the quick fix! This would have taken me ages to suspect that part of the code...
